e2e_tests/tests/cluster/test_ship_logs.py

```diff
@@ -0,0 +1,512 @@
+import io
+import json
+import logging
+import os
+import shutil
+import signal
+import socket
+import ssl
+import subprocess
+import sys
+import tempfile
+import textwrap
+import threading
+import time
+from typing import Any, List, Optional
+
+import pytest
+
+here = os.path.dirname(__file__)
+static_srv = os.path.join(here, "../../../master/static/srv")
+old = sys.path
+try:
+    sys.path = [static_srv] + sys.path
+    import ship_logs
+finally:
+    sys.path = old
+
+
+class ShipLogServer:
```

Comment:
I'm surprised this could be implemented so succinctly. Nice work.

Reply:
thank you